src/httpServer/handler.ts

```diff
@@ -13,21 +12,46 @@ import { Communicator } from "../communicator";
 import { APITypes, ArgSources } from "./handlerTypes";
 
 // local type declarations for workflow functions
-type WFFunc = (ctxt: WorkflowContext, ...args: any[]) => Promise<any>;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type WFFunc = (ctxt: WorkflowContext, ...args: any[]) => Promise<unknown>;
 export type InvokeFuncs<T> = WFInvokeFuncs<T> & AsyncHandlerWfFuncs<T>;
+export type InvokeFuncsInst<T> = WFInvokeFuncsInst<T>;
 
-export type AsyncHandlerWfFuncs<T> = {
-  [P in keyof T as T[P] extends WFFunc ? P : never]: T[P] extends WFFunc ? (...args: TailParameters<T[P]>) => Promise<WorkflowHandle<Awaited<ReturnType<T[P]>>>> : never;
-}
+export type AsyncHandlerWfFuncs<T> = 
+  T extends ConfiguredInstance
+  ? never
+  : {
+    [P in keyof T as T[P] extends WFFunc ? P : never]: T[P] extends WFFunc ? (...args: TailParameters<T[P]>) => Promise<WorkflowHandle<Awaited<ReturnType<T[P]>>>> : never;
+  };
 
-export type SyncHandlerWfFuncs<T> = {
-  [P in keyof T as T[P] extends WFFunc ? P : never]: T[P] extends WFFunc ? (...args: TailParameters<T[P]>) => Promise<Awaited<ReturnType<T[P]>>> : never;
-}
+export type SyncHandlerWfFuncs<T> =
+  T extends ConfiguredInstance
+  ? never
+  : {
+    [P in keyof T as T[P] extends WFFunc ? P : never]: T[P] extends WFFunc ? (...args: TailParameters<T[P]>) => Promise<Awaited<ReturnType<T[P]>>> : never;
```

Comment:
I know this was always there, but I think the `Promise<Awaited<whatever>>` is a no-op and can be condensed to just `whatever`.